cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorTest.java

```diff
@@ -352,6 +354,33 @@ public void testCreateClusterWithJmxEnabled(Params params, VertxTestContext cont
                 )); //getInitialCertificates(getKafkaAssembly("foo").getMetadata().getName()));
     }
 
+    @ParameterizedTest
+    @MethodSource("data")
+    public void testCreateClusterWithAnnotationsAndLabelsInCaCertSecret(Params params, VertxTestContext context) {
+        setFields(params);
+        Kafka kafka = getKafkaAssembly("foo");
+        kafka.getSpec()
+                .getKafka()
+                .setTemplate(new KafkaClusterTemplateBuilder()
+                        .withClusterCaCert(
+                                new ResourceTemplateBuilder()
+                                        .withNewMetadata()
+                                        .withLabels(Collections.singletonMap("foo", "bar"))
+                                        .withAnnotations(Collections.singletonMap("foo", "bar"))
+                                        .endMetadata()
+                                        .build())
+                        .build());
+
+        createCluster(context, kafka, Collections.singletonList(new SecretBuilder()
+                .withNewMetadata()
+                .withName(KafkaCluster.jmxSecretName("foo"))
+                .withNamespace("test")
+                .endMetadata()
+                .withData(Collections.singletonMap("foo", "bar"))
+                .build()
+        ));
+    }
```

Comment:
I'm not sure this works. First, it might run thousand times because of the parametrized test. So it would be better to add this to some of the existing tests and to the parameters. Also, this actually doesn't test the result. You set it up,  but you never compare the annotations and labels on the secrets.

I will think how to fix the test / or where to run it to make it better.